cfile/__init__.py

```diff
@@ -147,25 +144,27 @@ def append(self,elem):
    def extend(self,sequence):
       self.code.extend(sequence)
    def __str__(self):
+      head=str(self.head)+' ' if self.head is not None else ''
+      tail=str(self.tail) if self.tail is not None else ''

```

Comment:
If head ends with a space then tail should start with a space.
Now the code is inconsistent since you add the extra space for head at line 158 instead.
Remove space at line 158 and add it here instead.